kolibri/plugins/user/assets/src/actions.js

```diff
@@ -22,6 +23,60 @@ function showRoot(store) {
   });
 }
 
+function editProfile(store, edits, session) {
+  // payload needs username, fullname, and facility
+  // used to save changes to API
+  const savedUserModel = FacilityUserResource.getModel(session.user_id);
+  const changedValues = {};
+
+  // TODO set up core session updates
+
+  // explicit checks for the only values that can be changed
+  if (edits.full_name && edits.full_name !== session.full_name) {
+    changedValues.full_name = edits.full_name;
+  }
+  if (edits.username && edits.username !== session.username) {
+    changedValues.username = edits.username;
+  }
+  if (edits.password && edits.password !== session.password) {
+    changedValues.password = edits.password;
+  }
+
+  // check to see if anything's changed and conditionally add last requirement
+  if (Object.keys(changedValues).length) {
+    changedValues.facility = session.facility_id;
+  } else {
+    return;
+  }
```

Comment:
somewhat more direct:

```javascript
  // check to see if nothing's changed
  if (!Object.keys(changedValues).length) {
    return;
  }

  changedValues.facility = session.facility_id;
```

Also, since you're adding `facility` to the object (which stays the same), `changedValues` might be better named `userUpdateParams`.